_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -138,7 +269,8 @@ static const CodecId2Handlers codecId2Handlers =
             [](VideoCORE* core, mfxCoreInterface*, mfxStatus* mfxRes)
             -> VideoENCODE*
             {
-                return new MFXVideoENCODEMJPEG_HW(core, mfxRes);
+                *mfxRes = MFX_ERR_NONE;
+                return new FallbackProxyENCODE(core, MFX_CODEC_JPEG);
```

Comment:
don't do this in this patch. And you need to add ctors for HW/SW encoders. Create proxy class checking if both ctors present.